src/Rcl/Web/Templates.hs

```diff
@@ -84,28 +85,55 @@ resolverSelect name resolvers selected =
 
 changeTemplate :: Resolver -> PackageName -> Change -> Html ()
 changeTemplate resolver name = \case
-  Removed v -> packageSection "removed" resolver name v $ pure ()
-  Added v -> packageSection "added" resolver name v $ pure ()
+  Removed v -> packageSection "removed" resolver name (Nothing, v) $ pure ()
+  Added v -> packageSection "added" resolver name (Nothing, v) $ pure ()
   Changed fromV toV md -> do
     let cls = if toV > fromV then "upgraded" else "downgraded"
-    packageSection cls resolver name toV $ do
-      p_ $ toHtml $ "Previous version: " <> pack (showVersion fromV)
+    packageSection cls resolver name (Just fromV, toV) $ do
       unless (T.null md) $ changeLogDetails md
 
 packageSection
-  :: Text -> Resolver -> PackageName -> Version -> Html () -> Html ()
-packageSection cls resolver name v inner = section_ [class_ cls] $ do
-  p_ $ do
-    strong_ $ a_ [href_ packageUrl] $ toHtml packageName
-    toHtml $ " was " <> cls
-  inner
+  :: Text
+  -> Resolver
+  -> PackageName
+  -> (Maybe Version, Version)
+  -> Html ()
+  -> Html ()
+packageSection cls resolver name (mFromV, toV) inner =
+  section_ [class_ cls] $ do
+    p_ $ do
+      strong_ $ do
+        a_ [href_ packageUrl] $ do
+          toHtml $ unPackageName name
+      " " <> toHtml cls <> " "
+      strong_ $ versionChange mFromV toV
+    inner
  where
-  packageName = unPackageName name <> "-" <> pack (showVersion v)
   packageUrl =
     "https://www.stackage.org/"
       <> unResolver resolver
       <> "/package/"
-      <> packageName
+      <> unPackageName name
+      <> "-"
+      <> pack (showVersion toV)
+
+versionChange :: Maybe Version -> Version -> Html ()
+versionChange mFromV toV = do
```

Comment:
I worry about not having a direct unit test on this. I'd be afraid to touch it without breaking it, and loading up the site and spot-checking live examples is no fun.

I wonder, can an inner function at `Version -> Version -> SomethingStructured` be extracted and unit tested?

Reply:
Can totally do that.